### PR TITLE
[Levelset] Modifications to the algebraic stabilization

### PR DESCRIPTION
**Pull request by mrhashemi**
**📝 Description**
This PR addresses #9364 

Please mark the PR with appropriate tags: 
- Api Breaker, Mpi, etc...

**🆕 Changelog**
Please summarize the changes in one list to generate the changelog:
E.g.
- levelset_convection
- levelset_convection_element_simplex_algebraic_stabilization

@ddiezrod this will significantly improve the CFL limitation of the algebraic stabilization (with anti-diffusive terms). It is now possible to use CFL~1 with CN scheme.

### CODE REVIEW

**Comment by mrhashemi** [comment]
The new setting parameters (limiter acuteness) are highlighted:
{
            "model_part_name" : "",
            "echo_level" : 0,
            "convection_model_part_name" : "",
            "levelset_variable_name" : "DISTANCE",
            "levelset_convection_variable_name" : "VELOCITY",
            "levelset_gradient_variable_name" : "DISTANCE_GRADIENT",
            "max_CFL" : 1.0,
            "max_substeps" : 0,
            "eulerian_error_compensation" : false,
            **"BFECC_limiter_accuteness" : 2.0**,
            "element_type" : "levelset_convection_supg",
            "element_settings" : {
                "include_anti_diffusivity_terms" : false,
                **"elemental_limiter_accuteness" : 4.0**,
                "requires_distance_gradient" : false,
                "time_integration_theta" : 0.5
            }
}

File: kratos/processes/levelset_convection_process.h [hunk 1]
```diff
@@ -692,11 +697,11 @@ class LevelSetConvectionProcess
             const double fraction = std::abs(numerator) / (denominator + epsilon);
 
             if (mIsBfecc){
-                mLimiter[i_node] = 1.0 - std::pow(fraction, power_bfecc);
+                mLimiter[i_node] = 1.0 - std::pow(fraction, mPowerBfeccLimiter);
```
Comment (ddiezrod) [inline kratos/processes/levelset_convection_process.h:699]: One question: did you try without the limiter for BFECC ? I heard from other people in Altair (not Kratos developers) that they were quite happy with their results without having the limiter correction... Maybe it makes sense to have it as optional

Comment (mrhashemi) [reply kratos/processes/levelset_convection_process.h:699]: Without a limiter, BFECC disturbs the monotonicity of the underlying scheme. Having now the "acuteness" in the options, one can simply set a very large value to reduce the effect of the limiter (nearly deactivate it).

Comment (ddiezrod) [reply kratos/processes/levelset_convection_process.h:699]: Excuse my ignorance, but does the original equation (no BFECC and default "levelset_convection_element_simplex") satisfy the Discrete Maximum Principle? I thought it did not and that was why you could see over/under-shots. I am not very familiar with all these ideas, so any help is very welcome.

Comment (mrhashemi) [reply kratos/processes/levelset_convection_process.h:699]: The standard SUPG does not comply with DMP. The cross-wind stabilization improves it, however, it still is not monotonicity preserving (not full compliance with DMP). 
The point is that using BFECC without a limiter severely worsens the over/under-shoots.

Comment (ddiezrod) [reply kratos/processes/levelset_convection_process.h:699]: Thanks for the clarification 👍

File: kratos/processes/levelset_convection_process.h [hunk 2]
```diff
@@ -170,6 +170,7 @@ class LevelSetConvectionProcess
 
         // Evaluate steps needed to achieve target max_cfl
         const auto n_substep = EvaluateNumberOfSubsteps();
+        KRATOS_WATCH(n_substep)
```
Comment (rubenzorrilla) [inline kratos/processes/levelset_convection_process.h]: Seems a debugging leftover.
```suggestion
```

**Review by rubenzorrilla — APPROVED** [review]
For me this is ready :rocket: . Just for me to completely understand, basically this exposes two settings to the user and removes the linearisation of the limiter by adding the current step contribution to the RHS and LHS. Am I wrong?

**Comment by mrhashemi** [comment]
> For me this is ready 🚀 . Just for me to completely understand, basically this exposes two settings to the user and removes the linearisation of the limiter by adding the current step contribution to the RHS and LHS. Am I wrong?

1- Adding the optional limiter acuteness for BFECC and algebraic stabilization (separately).
2- Modify the contributions to the RHS and LHS for relaxing the limitation on CFL in the case of algebraic stabilization with the anti-diffusivity term (with theta > 0).

**Comment by ddiezrod** [comment]
@mrhashemi Do you want me to run the case I showed in https://github.com/KratosMultiphysics/Kratos/issues/9364 with these latest changes?

**Comment by mrhashemi** [comment]
> @mrhashemi Do you want me to run the case I showed in #9364 with these latest changes?

Yes, please without the "eulerian error compensation (BFECC)". You can now set CFL to a value around 0.8 (I assume) using theta = 0.5.

**Pull request merged**